Task40.Solution

LenousikK opened the pull request:
Added test classes for Task 40.

LenousikK committed:
Added:
1.Add implicit waiter for WebDriver.(Done)
2. Add Thread.sleep for login test, which was created on previous training. What type of waiter is it (add your answer as comment near sleep)? (Done)
3. Add explicit waiter for login test, which will wait until name appears (after login). Add polling frequency, which is differ from default value (Done)

LenousikK committed:
Added:
5. Create test with frames (URL - https://the-internet.herokuapp.com/iframe). Write the following text
   – Hello world! and check it. Do not use menu File -> New Document. Pay your attention that world
   must be with bold font (Done)

LenousikK committed:
Small corrections (re-naming of the variables) for test classes:
TestIFrame;
TestSimpleAlert
Added:
another assertion for TestSimpleAlert test class

LenousikK committed:
Revert "Small corrections (re-naming of the variables) for test classes: TestIFrame; TestSimpleAlert Added: another assertion for TestSimpleAlert test class"

This reverts commit 65914906

LenousikK committed:
1) Small corrections (re-naming of the variables) for test classes: TestIFrame;
2) Added test classes:
TestSimpleAlert;
TestConfirmAlert
This reverts commit 786d773ffb9d7e3e2641934f8864ff5ad8ff86af.

LenousikK committed:
Added:
6.Create 3 tests for alerts (URL - https://the-internet.herokuapp.com/javascript_alerts). (Done)
1) Small corrections (re-naming of the test method) in test class: TestConfirmAlert;
2) Added test class: TestPromptAlert.

LenousikK committed:
Code formatting for test class: TestPromptAlert.

LenousikK committed:
Added:
4. Create DDT test, which will login with following credentials:
seleniumtests@tut.by 123456789zxcvbn
seleniumtests2@tut.by 123456789zxcvbn
(Done)

EugenBorisik commented on src/test/java/LoginToTutBy.java:
```diff
@@ -37,13 +43,19 @@ public void closeBrowser() {
 
     @Test
-    public void LoginSuccessful() {
+    public void LoginSuccessful() throws InterruptedException {
```
Method name starts with small letter

EugenBorisik commented on src/test/java/TestConfirmAlert.java:
```diff
@@ -0,0 +1,44 @@
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.openqa.selenium.*;
```
Avoid asterisks in imports (i can setup IDEA to prevent auto-collapse)

EugenBorisik commented on src/test/java/TestConfirmAlert.java:
```diff
@@ -0,0 +1,44 @@
+    }
+
+    @Test
+    public void ConfirmAlert() {
```
Method name must start from small letter

EugenBorisik commented on src/test/java/TestIFrame.java:
```diff
@@ -0,0 +1,60 @@
+    }
+
+    @Test
+    public void TextIsEnteredToIFrame() {
```
Start with small

EugenBorisik commented on src/test/java/TestPromptAlert.java:
```diff
@@ -0,0 +1,47 @@
+    }
+
+    @Test
+    public void EnterValueInPromptAlert() {
```
Start with small

EugenBorisik commented on src/test/java/TestSimpleAlert.java:
```diff
@@ -0,0 +1,47 @@
+    }
+
+    @Test
+    public void ClickOnSimpleAlert() {
```
Start with small

EugenBorisik commented on src/test/java/TestIFrame.java:
```diff
@@ -0,0 +1,60 @@
+        driver.findElement(TEXT_AREA).sendKeys(TEXT_TO_ENTER_POINT);
+        String enteredText = driver.findElement(TEXT_AREA).getText();
+        String filteredEnteredText = "";
+        for (char character : enteredText.toCharArray()) {
```
I created such task with the following goal - sometimes there are some magic symbols/whitespaces/etc. and it should not block you (you can find workaround or even it is a bug)

EugenBorisik commented on src/test/java/TestSimpleAlert.java:
```diff
@@ -0,0 +1,47 @@
+            Alert alert = driver.switchTo().alert();
+            String textOnAlert = alert.getText();
+            alert.accept();
+            assertEquals("I am a JS Alert", textOnAlert);
```
I recommend to have 1 assertion per test, but there are situations like that. I such cases I prefer to use group assertions (assertAll). Please implement group assertion here

EugenBorisik reviewed (approved):
I close task, but please update code according my comments (all of them are related to namings + 1 recommendation)

LenousikK committed:
Corrected:
- method names start with small letter now;
- updated single assertions to group assertion in TestSimpleAlert test class
